src/org/pentaho/reporting/platform/plugin/output/FastXLSOutput.java

```diff
@@ -59,4 +76,41 @@ public boolean supportsPagination() {
   public Object getReportLock() {
     return this;
   }
+
+  public static void createXLS( final MasterReport report, final OutputStream outputStream,
```

Comment:
THere is a good case for removing that code here. Make the FastXLSOutput a subclasss of the normal XLSOutput and simply call "super.generate(..)" when you want to switch to the normal processing. 

Less code here means you have less new lines that Wingman complains about because of missing unit-tests :)